test/e2e/specs/editor/various/post-editor-template-mode.spec.js

```diff
@@ -378,4 +394,30 @@ class PostEditorTemplateMode {
 		);
 		await cancelButton.click();
 	}
+
+	/**
+	 * Edit the title of a template. The title is saved automatically, no
+	 * need to click anything, though we do click outside of the dialog
+	 * to close it.
+	 *
+	 * This needs to be called only after the template edit mode has been
+	 * activated or it will fail, i.e after creating a new custom template
+	 * or after calling `switchToTemplateMode()`.
+	 *
+	 * @param {string} newTitle the new title to set.
+	 * @return {Promise<void>}
+	 */
+	async editTemplateTitle( newTitle ) {
+		await this.page.click( 'button[aria-label="Template Options"]' );
+
+		const editDialog = this.page.locator(
+			'div.edit-site-template-details__group'
```

Comment:
Could we rewrite it into a role selector instead?

Reply:
I haven't used `getByRole` that much in the past, so I don't know its heuristics from the top of my head, but possibly, yeah. . I will have a deeper look at the markup to verify.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/50195/commits/c3808a3717133b6d71b03ea40d3e08a8340dcb77.